src/hooks/useBuyout/index.ts

```diff
@@ -0,0 +1,120 @@
+import { isEmpty } from 'lodash';
+import { Option } from '@polkadot/types-codec';
+import { Codec } from '@polkadot/types-codec/types';
+
+import { useNodeInfoState } from '../../NodeInfoProvider';
+import { nativeToFormatDecimalPure } from '../../shared/parseNumbers/decimal';
+import { doSubmitExtrinsic } from '../../pages/collators/dialogs/helpers';
+import { useGlobalState } from '../../GlobalStateProvider';
+
+import { OrmlTraitsAssetRegistryAssetMetadata } from './types';
+import { getMetadata, scaleByCurrencyPrecision } from './utils';
+import { ToastMessage, showToast } from '../../shared/showToast';
+import { PerMill } from '../../shared/parseNumbers/permill';
+
+export interface BuyoutSettings {
+  buyoutNativeToken: {
+    min: number;
+    max: number;
+  };
+  currencies: OrmlTraitsAssetRegistryAssetMetadata[];
+  nativeCurrency: OrmlTraitsAssetRegistryAssetMetadata;
+  handleBuyout: (
+    currency: OrmlTraitsAssetRegistryAssetMetadata,
+    amount: number,
+    setSubmissionPending: StateUpdater<boolean>,
+    setConfirmationDialogVisible: StateUpdater<boolean>,
+  ) => void;
+}
+
+function handleBuyoutError(error: string) {
+  if (error.includes('1010: Invalid Transaction: Custom error: 3')) {
+    showToast(ToastMessage.ERROR, 'The buyout amount is too low');
+    return;
+  }
+
+  if (error.includes('Cancelled')) {
+    showToast(ToastMessage.ERROR, 'The buyout was cancelled');
+    return;
+  }
+
+  showToast(ToastMessage.ERROR, 'The buyout failed');
+  return;
+}
+
+export const useBuyout = (): BuyoutSettings => {
+  const { api } = useNodeInfoState().state;
+  const { walletAccount } = useGlobalState();
+  const [minimumBuyout, setMinimumBuyout] = useState<number>(0);
+  const [maximumBuyout, setMaximumBuyout] = useState<number>(0);
+
+  useEffect(() => {
+    async function fetchBuyoutLimits() {
+      if (api) {
+        const minBuyoutLimitFromAPI = api.consts.treasuryBuyoutExtension.minAmountToBuyout;
+        const maxBuyoutLimitFromAPI = await api.query.treasuryBuyoutExtension.buyoutLimit();
+        const sellFeeFromAPI = api.consts.treasuryBuyoutExtension.sellFee;
+
+        const minBuyoutLimitAsString = minBuyoutLimitFromAPI?.toString();
+        const maxBuyoutLimitAsString = (maxBuyoutLimitFromAPI as unknown as Option<Codec>)?.unwrap().toString();
+        const sellFeeAsString = sellFeeFromAPI?.toString();
+
+        const minBuyoutLimitFormatted = nativeToFormatDecimalPure(minBuyoutLimitAsString);
+        const maxBuyoutLimitFormatted = nativeToFormatDecimalPure(maxBuyoutLimitAsString);
+        const sellFeePerMill = new PerMill(Number(sellFeeAsString));
+
+        const MIN_FEE_MULTIPLIER = 1.01; // sellFee + min amount still needs 1% for the tx to be fired.
+
+        setMinimumBuyout(sellFeePerMill.applyAdjustmentToNumber(minBuyoutLimitFormatted) * MIN_FEE_MULTIPLIER);
+        setMaximumBuyout(sellFeePerMill.applyAdjustmentToNumber(maxBuyoutLimitFormatted));
```

Comment:
Maybe we just need to refactor these calculations. Let's try to use "big.js" numbers for increased accuracy and let's also try to calculate on the raw value instead of the decimal one. 

I.e. I would change it so that we first instantiate big numbers (using "big.js") of the raw min and max values and then multiply that with the fee. The end result would be something like
```
        // use big.js
        const minBuyoutLimit = new BigNumber(minBuyoutLimitAsString);
        minBuyoutLimit.mul(sellFeePerMill.toDecimal());
```

Not sure if this then works without this `MIN_FEE_MULTIPLIER` but it's worth a shot.

Reply:
It doesn't work unfortunately but we can implement to use `"buyout ": { "amount": xxx }` also for the min buyout case - WDYT?

Reply:
Yes, let's do that then 👍